error installing aglio on windows7

Hello All,

Although being a linux fan, my company's requirement is to work under windows. I'm trying to install agliounder windows and I'm failing with it:

```
$ npm install -g aglio
npm WARN optional dep failed, continuing fsevents@1.0.5

> utf-8-validate@1.2.1 install C:\Program Files\nodejs\node_modules\aglio\node_modules\socket.io\node_modules\socket.io-client\node_modules\engine.io-client\node_modules\ws\node_modules\utf-8-validate
> node-gyp rebuild


C:\Program Files\nodejs\node_modules\aglio\node_modules\socket.io\node_modules\socket.io-client\node_modules\engine.io-client\node_modules\ws\node_modules\utf-8-validate>node "C:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild
Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.
MSBUILD : error MSB3428: Could not load the Visual C++ component "VCBuild.exe". To fix this, 1) install the .NET Framework 2.0 SDK, 2) install Microsoft Visual Studio 2005 or 3) add the location of the component to the system path if it is installed elsewhere.  [C:\Program Files\nodejs\node_modules\aglio\node_modules\socket.io\node_modules\socket.io-client\node_modules\engine.io-client\node_modules\ws\node_modules\utf-8-validate\build\binding.sln]
gyp
> utf-8-validate@1.2.1 install C:\Program Files\nodejs\node_modules\aglio\node_modules\socket.io\node_modules\engine.io\node_modules\ws\node_modules\utf-8-validate
> node-gyp rebuild


C:\Program Files\nodejs\node_modules\aglio\node_modules\socket.io\node_modules\engine.io\node_modules\ws\node_modules\utf-8-validate>node "C:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild
Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.
MSBUILD : error MSB3428: Could not load the Visual C++ component "VCBuild.exe". To fix this, 1) install the .NET Framework 2.0 SDK, 2) install Microsoft Visual Studio 2005 or 3) add the location of the component to the system path if it is installed elsewhere.  [C:\Program Files\nodejs\node_modules\aglio\node_modules\socket.io\node_modules\engine.io\node_modules\ws\node_modules\utf-8-validate\build\binding.sln]
gypnpm
> bufferutil@1.2.1 install C:\Program Files\nodejs\node_modules\aglio\node_modules\socket.io\node_modules\socket.io-client\node_modules\engine.io-client\node_modules\ws\node_modules\bufferutil
> node-gyp rebuild

 WARN optional dep failed, continuing utf-8-validate@1.2.1

C:\Program Files\nodejs\node_modules\aglio\node_modules\socket.io\node_modules\socket.io-client\node_modules\engine.io-client\node_modules\ws\node_modules\bufferutil>node "C:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild
Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.
MSBUILD : error MSB3428: Could not load the Visual C++ component "VCBuild.exe". To fix this, 1) install the .NET Framework 2.0 SDK, 2) install Microsoft Visual Studio 2005 or 3) add the location of the component to the system path if it is installed elsewhere.  [C:\Program Files\nodejs\node_modules\aglio\node_modules\socket.io\node_modules\socket.io-client\node_modules\engine.io-client\node_modules\ws\node_modules\bufferutil\build\binding.sln]
gypnpm
> bufferutil@1.2.1 install C:\Program Files\nodejs\node_modules\aglio\node_modules\socket.io\node_modules\engine.io\node_modules\ws\node_modules\bufferutil
> node-gyp rebuild

 WARN optional dep failed, continuing utf-8-validate@1.2.1

C:\Program Files\nodejs\node_modules\aglio\node_modules\socket.io\node_modules\engine.io\node_modules\ws\node_modules\bufferutil>node "C:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild
Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.
MSBUILD : error MSB3428: Could not load the Visual C++ component "VCBuild.exe". To fix this, 1) install the .NET Framework 2.0 SDK, 2) install Microsoft Visual Studio 2005 or 3) add the location of the component to the system path if it is installed elsewhere.  [C:\Program Files\nodejs\node_modules\aglio\node_modules\socket.io\node_modules\engine.io\node_modules\ws\node_modules\bufferutil\build\binding.sln]
gypnpm
> protagonist@1.1.1 install C:\Program Files\nodejs\node_modules\aglio\node_modules\protagonist
> node-gyp rebuild

 WARN optional dep failed, continuing bufferutil@1.2.1
npm WARN engine hoek@2.16.3: wanted: {"node":">=0.10.40"} (current: {"node":"0.10.35","npm":"2.5.1"})
npm WARN engine boom@2.10.1: wanted: {"node":">=0.10.40"} (current: {"node":"0.10.35","npm":"2.5.1"})
npm WARN engine cryptiles@2.0.5: wanted: {"node":">=0.10.40"} (current: {"node":"0.10.35","npm":"2.5.1"})

C:\Program Files\nodejs\node_modules\aglio\node_modules\protagonist>node "C:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild
Warning: Missing input files:
C:\Program Files\nodejs\node_modules\aglio\node_modules\protagonist\build\drafter\ext\snowcrash\..\..\..\..\drafter\ext\snowcrash\src\Version.h
Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.
MSBUILD : error MSB3428: Could not load the Visual C++ component "VCBuild.exe". To fix this, 1) install the .NET Framework 2.0 SDK, 2) install Microsoft Visual Studio 2005 or 3) add the location of the component to the system path if it is installed elsewhere.  [C:\Program Files\nodejs\node_modules\aglio\node_modules\protagonist\build\binding.sln]
MSBUILD : error MSB3428: Could not load the Visual C++ component "VCBuild.exe". To fix this, 1) install the .NET Framework 2.0 SDK, 2) install Microsoft Visual Studio 2005 or 3) add the location of the component to the system path if it is installed elsewhere.  [C:\Program Files\nodejs\node_modules\aglio\node_modules\protagonist\build\binding.sln]
MSBUILD : error MSB3428: Could not load the Visual C++ component "VCBuild.exe". To fix this, 1) install the .NET Framework 2.0 SDK, 2) install Microsoft Visual Studio 2005 or 3) add the location of the component to the system path if it is installed elsewhere.  [C:\Program Files\nodejs\node_modules\aglio\node_modules\protagonist\build\binding.sln]
MSBUILD : error MSB3428: Could not load the Visual C++ component "VCBuild.exe". To fix this, 1) install the .NET Framework 2.0 SDK, 2) install Microsoft Visual Studio 2005 or 3) add the location of the component to the system path if it is installed elsewhere.  [C:\Program Files\nodejs\node_modules\aglio\node_modules\protagonist\build\binding.sln]
MSBUILD : error MSB3428: Could not load the Visual C++ component "VCBuild.exe". To fix this, 1) install the .NET Framework 2.0 SDK, 2) install Microsoft Visual Studio 2005 or 3) add the location of the component to the system path if it is installed elsewhere.  [C:\Program Files\nodejs\node_modules\aglio\node_modules\protagonist\build\binding.sln]
gypnpm
> aglio-theme-olio@1.6.2 postinstall C:\Program Files\nodejs\node_modules\aglio\node_modules\aglio-theme-olio
> node scripts/setup-cache.js

 WARN optional dep failed, continuing bufferutil@1.2.1
npm ERR! Windows_NT 6.1.7601
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install" "-g" "aglio"
npm ERR! node v0.10.35
npm ERR! npm  v2.5.1
npm ERR! code ELIFECYCLE

npm ERR! protagonist@1.1.1 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the protagonist@1.1.1 install script 'node-gyp rebuild'.
npm ERR! This is most likely a problem with the protagonist package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls protagonist
npm ERR! There is likely additional logging output above.
```

I can see that agliois trying to use something like MSBUILD, which - as I found in www - is related to MS Visual Studio. My questions are:
* am I doing something wrong? Maybe I need to install something prior to aglio?
* is there any possibility to install aglio on windows without having MS Visual Studio?

--- a/package.json
+++ b/package.json
@@ -39,7 +39,7 @@
     "chokidar": "^1.4.1",
     "cli-color": "^1.1.0",
     "pretty-error": "^1.2.0",
-    "protagonist": "^1.3.2",
+    "drafter": "^1.0.0",
     "serve-static": "^1.10.0",
     "socket.io": "^1.3.7",
     "yargs": "^3.31.0"

--- a/src/main.coffee
+++ b/src/main.coffee
@@ -1,6 +1,6 @@
 fs = require 'fs'
 path = require 'path'
-protagonist = require 'protagonist'
+drafter = require 'drafter'
 
 INCLUDE = /( *)<!-- include\((.*)\) -->/gmi
 ROOT = path.dirname __dirname
@@ -82,7 +82,7 @@ exports.render = (input, options, done) ->
     # Handle custom directive(s)
     input = includeDirective options.includePath, input
 
-    # Protagonist does not support \r ot \t in the input, so
+    # Drafter does not support \r ot \t in the input, so
     # try to intelligently massage the input so that it works.
     # This is required to process files created on Windows.
     filteredInput = if not options.filterInput then input else
@@ -91,7 +91,7 @@ exports.render = (input, options, done) ->
             .replace(/\t/g, '    ')
 
     benchmark.start 'parse'
-    protagonist.parse filteredInput, type: 'ast', (err, res) ->
+    drafter.parse filteredInput, type: 'ast', (err, res) ->
         benchmark.end 'parse'
         if err
             err.input = input

--- a/test/basic.coffee
+++ b/test/basic.coffee
@@ -4,7 +4,7 @@ bin = require '../lib/bin'
 fs = require 'fs'
 http = require 'http'
 path = require 'path'
-protagonist = require 'protagonist'
+drafter = require 'drafter'
 sinon = require 'sinon'
 
 root = path.dirname(__dirname)
@@ -164,13 +164,13 @@ describe 'API Blueprint Renderer', ->
             done()
 
     it 'Should error on drafter failure', (done) ->
-        sinon.stub protagonist, 'parse', (content, options, callback) ->
+        sinon.stub drafter, 'parse', (content, options, callback) ->
             callback 'error'
 
         aglio.render blueprint, 'default', (err, html) ->
             assert err
 
-            protagonist.parse.restore()
+            drafter.parse.restore()
 
             done()
 

